File: 2796_repeat_string/Solution.scala
// LeetCode 2796 - Repeat String
// https://leetcode.com/problems/repeat-string/

object Solution {
  def replicate(str: String, times: Int): String = {
    if (times <= 0) return ""
    val sb = new StringBuilder(str.length * times)
    var i = 0
    while (i < times) {
      sb.append(str)
      i += 1
    }
    sb.toString
  }
}

File: 2796_repeat_string/Solution.kt
// LeetCode 2796 - Repeat String
// https://leetcode.com/problems/repeat-string/
// JS-only problem; C# stand-in.

class Solution {
    fun replicate(str: String, times: Int): String {
        if (times <= 0) return ""
        var sb = StringBuilder(str.length * times)
        for (i in 0 until times) { sb.append(str) }
        return sb.toString()
    }
}
